oauth.go

```diff
@@ -24,6 +25,25 @@ type Host struct {
 	TokenURL      string
 }
 
+func NewGitHubHost(hostURL string) (*Host, error) {
```

Comment:
Let's add a Go doc comment since this is an exported function. 

```suggestion

// NewGitHubHost constructs a Host from the given URL to a GitHub instance.
func NewGitHubHost(hostURL string) (*Host, error) {
```